src/basic/sparse-endian.h

```diff
@@ -55,19 +54,19 @@ typedef uint64_t __sd_bitwise be64_t;
 #undef le64toh
 
 #if __BYTE_ORDER == __LITTLE_ENDIAN
-#define bswap_16_on_le(x) bswap_16(x)
-#define bswap_32_on_le(x) bswap_32(x)
-#define bswap_64_on_le(x) bswap_64(x)
```

Comment:
sorry, but this is wrong. byteswap.h aleady does the very same thing internally for it's implementation of bswap_32(). hence this is enirely unnecessary.

generally we have the rule that we'll use compiler extensions where it makes sense, but we'd rather use libc wrappers if available.